segway/_util.py

```diff
@@ -189,11 +189,15 @@ def get_col_index(chromosome, trackname):
 
 
 def get_label_color(label):
-    color = SCHEME[label % NUM_COLORS]
+    color = SCHEME[hash_label(label) % NUM_COLORS]
```

Comment:
Maybe there should be some data structure that holds some state and counts the current usage/collision count. So you can simply just cycle through the colors based on usage numbers.

Reply:
Can you elaborate on this? I don't understand what the "current usage/collision" count means.
Also because each process has a subsection of the genome, this structure would have to continue working even if the count is reset (a new process starts on a new section) or not all labels values are known (while the number of labels is known, their values are not passed).

Reply:
The idea is to avoid collisions by simply moving to the next available slot. E.g.
```python
max_collision_count = max([SCHEME_USAGE[key] for key in SCHEME_USAGE])
current_key = label % colors
if SCHEME_USAGE[current_key] >0 and SCHEME_USAGE[current_key] <= max_collision_count:
  # iterate through remaining table until you find an entry with <= max
  # put into "new key"
  SCHEME_USAGE[new_key] += 1 # update usage count
color = SCHEME[new_key]
```
Something along those lines but probably better thought out.

Reply:
I still don't understand how this code avoids collisions and don't see how this solves the coloring problem, so we should discuss it at the meeting. 

On this specific code, I don't think it ensures consistent coloring. Suppose labels A and B have the same key and there are two subprocesses: one over a region with label A over 75% of its length and label B over the last 25%, and one over a region with label B over 75% of its length and label A over the last 25%. These subprocesses effectively operate independently (as the information of labels and collisions in other regions is not shared between subprocesses), so by symmetry the first subprocess should assign A to `current_key` and B to `new_key` but the second subprocess should assign B to `current_key` and A to `new_key`. I might be misunderstanding the algorithm, so let me know if there's a workaround at the meeting.